File: day1.js
import { readInput } from "./helpers/fileReader.js";


console.log("JS")
const expense = readInput("./input/day1.txt");
//find which 2 items add up to 2020

for (let i = 0; i < expense.length; i++) {
  let num1 = parseInt(expense[i]);
  for (let j = i + 1; j < expense.length; j++) {
    let num2 = parseInt(expense[j]);
    if (num1 + num2 === 2020) {
      console.log("Part 1 :", num1 * num2);
    }
  }
}

for (let i = 0; i < expense.length; i++) {
  for (let j = i + 1; j < expense.length; j++) {
    for (let k = j + 1; k < expense.length; k++) {
      if (
        parseInt(expense[i]) + parseInt(expense[j]) + parseInt(expense[k]) ===
        2020
      ) {
        console.log(
          "Part 2 :",
          parseInt(expense[i]) * parseInt(expense[j]) * parseInt(expense[k])
        );
      }
    }
  }
}

File: day1.py
file = open('./input/day1.txt')
file = [x.strip() for x in file.readlines()]
print("Python")


def part1():
    for index, record in enumerate(file):
        num1 = int(record)
        for j in file[index:]:
            num2 = int(j)
            if (num1 + num2 == 2020):
                print('Part 1 : {}'.format(num1 * num2))


def part2():
    breakFlag = 0
    for index, record1 in enumerate(file):
        num1 = int(record1)
        for record2 in file[index:]:
            num2 = int(record2)
            num3 = 2020 - num1 - num2
            if str(num3) in file:
                print('Part 2 : {}'.format(num3 * num1 * num2))
                breakFlag = 1
                break
        if breakFlag:
            break


part1()
part2()